src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -321,107 +300,107 @@ private bool TryLookupLengthOrCount(SyntaxNode node, TypeSymbol inputType, out P
             }
             else
             {
-                elementType = CreateErrorType();
-                isUnsupported = true;
-            }
-
-            // We may have looked up the length property for the length pattern.
-            if (lengthProperty is null && !TryLookupLengthOrCount(node, inputType, out lengthProperty))
-                isUnsupported = true;
-
-            if (isUnsupported && !inputType.IsErrorType())
-            {
-                // Report an unsupported type if either the length property or a viable indexer is missing.
-                Error(diagnostics, ErrorCode.ERR_UnsupportedTypeForListPattern, node, inputType);
                 hasErrors = true;
+                if (!inputType.IsErrorType())
```

Comment:
Consider moving `inputType.IsErrorType()` to be a top-level case for the main `if` of this method.